grails-app/controllers/org/pih/warehouse/inventory/InventoryController.groovy

```diff
@@ -575,49 +575,70 @@ class InventoryController {
 
 
     def listExpiredStock() {
-        def location = Location.get(session.warehouse.id)
-        def categorySelected = (params.category) ? Category.get(params.category) : null
-        def inventoryItems = dashboardService.getExpiredStock(categorySelected, location)
-        def categories = inventoryItems?.collect { it.product.category }?.unique()
-        def quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(location)
-        def expiredStockMap = [:]
-        inventoryItems.each { inventoryItem ->
-            expiredStockMap[inventoryItem] = quantityMap[inventoryItem]
+        Location location = Location.get(session.warehouse.id)
+        Category categorySelected = Category.get(params.category)
+        Boolean withBinLocation = params.boolean("withBinLocation")
+
+        List<InventoryItem> inventoryItems = dashboardService.getExpiredStock(categorySelected, location)
+        List<Category> categories = inventoryItems?.collect { it.product.category }?.unique()
+
+        List<Map> data
+        if (withBinLocation) {
+            data = productAvailabilityService.getQuantityOnHandByBinLocation(location)
+        } else {
+            data = productAvailabilityService.getQuantityOnHandByInventoryItem(location)
+                    .collect{ key, val -> [ inventoryItem: key, quantity: val ] }
         }
```

Comment:
Can't it be just written using a ternary operator?

Reply:
sure, I can.
I see you guys really hate if/else statements... :sweat_smile: 
I feel like simple if/else most of the time is much more readable than a ternary.
Ternary is great for very short statements and this case kind of applies to that, although I was also considering the case where in the future we want to add another parameter which would call another service method and ternary in this case would be very bad.

But yeah, I can change it to ternary.

Reply:
from my perspective it was only a suggestion for you to decide, which approach you prefer there. I want everyone to feel comfortable with their code style, since your version is not "breaking" any rules.

Reply:
No @kchelstowski that is not how this works. It's ternary or nothing!!!!

Reply:
Just kidding. I don't feel strongly either way. I do feel strongly that the code should be refactored and probably moved to the service layer and that the getExpiredStock should probably handle all or most of the work. But we can save that for another day in case it'll be too much work.